### DISCUSSION TITLE
add cypress component test in jenkins pipeline

### DISCUSSION THREAD

**Discussion by vahidsamimi** [cypress-io/cypress#26176] [Questions and Help]
I want to add a new stage to run Cypress component tests in a Jenkins pipeline. I have installed Cypress 12.7 and added it to the package.json file of my Angular project to run Cypress like this:

`"cypress:run": " cypress run --component  --browser chrome --headless"`
in jenkinsfile added the new stage like that:

```
stage('Cypress Test') {
            environment {
                ELECTRON_RUN_AS_NODE = 1
              }
            steps {
                dir("${ANGULAR_WORKSPACE}") {
                    sh 'npm run cypress:run'
                }
            }
        }
```
First, I ran it without `--headless` parameter, but I got an error that I should install plugin Xvfb

```
It looks like this is your first time using Cypress: 3.4.0

[?25l[01:56:21]  Verifying Cypress can run /var/lib/jenkins/.cache/Cypress/3.4.0/Cypress [started]
[01:56:21]  Verifying Cypress can run /var/lib/jenkins/.cache/Cypress/3.4.0/Cypress [failed]
[?25hYour system is missing the dependency: Xvfb

Install Xvfb and run Cypress again.

Read our documentation on dependencies for more information:

https://on.cypress.io/required-dependencies

If you are using Docker, we provide containers with all required dependencies installed.

----------

Error: spawn Xvfb ENOENT

----------
````
after read the [cypress doc](https://docs.cypress.io/guides/continuous-integration/introduction#Running-headless-tests-without-Xvfb), I added the env variable `ELECTRON_RUN_AS_NODE = 1` in cypress stage in jenkinsfile.

but now I get the following error when the pipeline starts to run:

```
> cypress run --component  --browser chrome --headless

It looks like this is your first time using Cypress: 12.7.0

[STARTED] Task without title.
[FAILED] Cypress failed to start.
[FAILED] 
[FAILED] This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies
[FAILED] Please refer to the error below for more details.
[FAILED] /home/insure/.cache/Cypress/12.7.0/Cypress/Cypress: error while loading shared libraries: libdrm.so.2: cannot open shared object file: No such file or directory
[FAILED] Platform: linux-x64 (Red Hat Linux - 8.4)
[FAILED] Cypress Version: 12.7.0

Cypress failed to start.
This may be due to a missing library or dependency. https://on.cypress.io/required-dependencies
Please refer to the error below for more details.

/home/insure/.cache/Cypress/12.7.0/Cypress/Cypress: error while loading shared libraries: libdrm.so.2: cannot open shared object file: No such file or directory

Platform: linux-x64 (Red Hat Linux - 8.4)
Cypress Version: 12.7.0

npm ERR! Lifecycle script `cypress:run` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: workplace@0.0.0 
npm ERR!   at location: /home/insure/workspace/INSWFWP-6716_cypress_in_pipeline/workplace 
script returned exit code 1
```

**Comment by portyu9** [comment]
Hi @vahidsamimi 

The current error is an **OS dependency problem**, not a Component Testing problem:

```text
error while loading shared libraries: libdrm.so.2:
cannot open shared object file: No such file or directory
```

Cypress's binary still requires native Linux libraries even when you run:

```bash
--headless
```

On RHEL-like systems, `libdrm.so.2` is normally provided by the `libdrm` package. You should also check the Cypress executable for any other missing libraries rather than fixing them one at a time:

```bash
ldd ~/.cache/Cypress/12.7.0/Cypress/Cypress | grep 'not found'
```

I would also remove:

```bash
ELECTRON_RUN_AS_NODE=1
```

That is not a supported substitute for installing Cypress's operating-system dependencies.

There is another clue in the original logs: one invocation reports **Cypress 3.4.0**, while your package is supposed to use **12.7.0**. Verify what Jenkins is actually resolving:

```bash
npx cypress version
npx cypress cache list
```

and install dependencies from a clean workspace with:

```bash
npm ci
```

For a current setup, the cleaner solution is to run the Jenkins test stage in an **official Cypress Docker image**, which already contains the necessary Linux libraries and browsers. Current Cypress also no longer lists RHEL 8.4 as a supported host OS.

So the immediate failure is missing `libdrm.so.2`; the longer-term fix is to use a supported environment with all Cypress prerequisites installed.

Please mark this answer as accepted if it helped, thank you.
